packages/block-library/src/block/edit.js

```diff
@@ -77,6 +143,40 @@ export default function ReusableBlockEdit( {
 			: InnerBlocks.ButtonBlockAppender,
 	} );
 
+	const subRegistry = useMemo( () => {
+		return {
+			...registry,
+			dispatch( store ) {
+				if (
+					store !== blockEditorStore &&
+					store !== blockEditorStore.name
+				) {
+					return registry.dispatch( store );
+				}
+				const dispatch = registry.dispatch( store );
+				const select = registry.select( store );
+				return {
+					...dispatch,
+					updateBlockAttributes(
+						clientId,
+						attributes,
+						uniqueByBlock
+					) {
+						return updateBlockAttributes( patternClientId )(
+							clientId,
+							attributes,
+							uniqueByBlock
+						)( {
+							registry,
+							select,
+							dispatch,
+						} );
+					},
+				};
+			},
+		};
+	}, [ registry, patternClientId ] );
```

Comment:
@gziolo This is another alternative to https://github.com/WordPress/gutenberg/pull/54233, which nests `registry` to monkey-patch `updateBlockAttributes` for only a given sub-tree. This only solves the "dispatching" side of things too though, for selecting it's still getting the attributes from the HOC. Just want to get a quick check from you or anyone on what you think of this approach? Personally, even though it's still monkey-patching which could seem a little bit hacky, I think this is still better than patching `updateBlockAttributes` directly.

Reply:
Thank you for sharing it. That’s a pretty interesting idea because it would potentially limit the impact of adding additional logic to `updateBlockAttributes`. If paired with the selector that returns the list of attributes for the block, we would have quite a robust solution. Have you measured the performance impact on typing in general when applying changes to `updateBlockAttributes` in the original block editor store in the previous prototype? I’m curious what’s the rationale behind using the nested registry. One note, I see there is `useEntityBlockEditor` used in the block so there might be already a nested registry present.